cpp/src/arrow/type_traits.h

```diff
@@ -368,6 +368,49 @@ struct TypeTraits<ExtensionType> {
   constexpr static bool is_parameter_free = false;
 };
 
+namespace internal {
+
+template <typename... Ts>
+struct make_void {
+  using type = void;
+};
+
+template <typename... Ts>
+using void_t = typename make_void<Ts...>::type;
```

Comment:
could you use `as_void` instead?
https://github.com/apache/arrow/pull/5171/files#diff-1bd9192beb3fba5a1140dbd38b88e729R551

Reply:
(or replace as_void with void_t)

Reply:
Okay. Replacing `as_void` with `void_t` as it takes variadic arguments.